File: src/main/java/vip/yazilim/libs/springcore/rest/model/RestResponse.java
package vip.yazilim.libs.springcore.rest.model;

import lombok.Data;

/**
 * Generic Response model for REST Microservices.
 * 
 * @author Emre Sen, 24.07.2019
 * @contact maemresen@yazilim.vip
 */
@Data
public class RestResponse<E> {

    private long timestamp;
    private String path;
    private final boolean hasError;
    private String message;

    private E data;

    public RestResponse() {
        this(false);
    }

    public RestResponse(boolean hasError) {
        this.hasError = hasError;
    }

}

File: src/main/kotlin/vip/yazilim/libs/springvip/util/generic/rest/model/RestResponse.kt
package vip.yazilim.libs.springvip.util.generic.rest.model

/**
 * Generic Response model for REST Microservices.
 *
 * @author Emre Sen (maemresen@yazilim.vip), 24.07.2019
 */
open class RestResponse<E>(val hasError: Boolean = false
                           , val timestamp: Long
                           , val path: String
                           , val message: String
                           , val data: E)
